content/blog/2016/2016-08-09-ewm-beta-version.adoc

```diff
@@ -0,0 +1,173 @@
+
+==== Multiple upstream run selection strategies
+It has support for the
+link:https://github.com/jenkinsci/run-selector-plugin[Run Selector Plugin] (which is still in beta),
+so you can provide different run selection strategies when allocating a disk from the upstream job.
+
+Let's suppose that we have an upstream job that clones the repository and builds the project:
+```groovy
+def extWorkspace = exwsAllocate 'diskpool1'
+
+node ('linux') {
+    exws (extWorkspace) {
+        checkout scm
+        sh 'mvn clean install -DskipTests'
+    }
+}
+```
+
+In the downstream job, we run the tests on a different node, but we reuse the same workspace as the previous job:
+```groovy
+def run = selectRun 'upstream'
+def extWorkspace = exwsAllocate selectedRun: run
+
+node ('test') {
+    exws (extWorkspace) {
+        sh 'mvn test'
+    }
+}
+```
+
+The `selectRun` in this example selects the last stable build from the upstream job.
+But, we can be more explicit, and select a specific build number from the upstream job.
+```groovy
+def run = selectRun 'upstream',
+ selector: [$class: 'SpecificRunSelector', buildNumber: UPSTREAM_BUILD_NUMBER]
+def extWorkspace = exwsAllocate selectedRun: run
+// ...
+```
+When the `selectedRun` parameter is given to the `exwsAllocate` step, it will allocate the same workspace that was
+used by that run.
+
+The Run Selector Plugin has several run selection strategies that are briefly explained
+link:https://github.com/jenkinsci/run-selector-plugin/blob/master/README.md[here].
+
+==== Automatic workspace cleanup
+
+Provides an automatic workspace cleanup by integrating the
+link:https://github.com/jenkinsci/ws-cleanup-plugin[Workspace Cleanup Plugin].
+For example, if we need to delete the workspace only if the build has failed, we can do the following:
+
+```groovy
+def extWorkspace = exwsAllocate diskPoolId: 'diskpool1'
+
+node ('linux') {
+    exws (extWorkspace) {
+        try {
+            checkout scm
+            sh 'mvn clean install'
+        } catch (e) {
+            currentBuild.result = 'FAILURE'
+            throw e
+        } finally {
+            step ([$class: 'WsCleanup', cleanWhenFailure: false])
+        }
+    }
+}
+```
+
+More workspace cleanup examples can be found at this
+link:https://github.com/jenkinsci/external-workspace-manager-plugin/blob/master/doc/WORKSPACE_CLEANUP.md[link].
+
+==== Custom workspace path
+
+Allows the user to specify a custom workspace path to be used when allocating workspace on the disk.
+The plugin offers two alternatives for doing this:
+
+* by defining a global workspace template for each Disk Pool
+
+This can be defined in the Jenkins global config, __External Workspace Definitions__ section.
+
+image:/images/post-images/ewm/global-custom-workspace-path.png[title="Global Custom Workspace Path", role="center"]
+
+* by defining a custom workspace path in the Pipeline script
+
+We can make use of the Pipeline DSL to compute the workspace path as we want.
```

Comment:
Not important: We can use the Pipeline DSL to compute the workspace path. (no need to say "make use of", you can just say "use" since we are talking about the Pipeline DSL in the current context; the verb "compute" implies that it will be "as we want").